worker/deployer/nested.go

```diff
@@ -37,6 +38,12 @@ type Logger interface {
 	Tracef(string, ...interface{})
 }
 
+// RebootMonitorStatePurger is implemented by types that can clean up the
```

Comment:
Nice catch! I wonder why these changes got left out in the refactoring since they were probably on both 2.8 and develop by the time the agent unification bits landed. Could this be a copy of the original file that lived in the refactoring branch? That might explain why the changes got lost...